code/modules/spells/spell_types/hivemind.dm

```diff
@@ -884,91 +500,337 @@
+			if(3)
+				user.visible_message("<span class='danger'>[target] begins to look frantic!</span>", "<span class='notice'>We begin to override [target]'s conciousness with our own.</span>")
+				to_chat(target, "<span class='userdanger'>Your conciousness beings to waver!</span>")
+
+		if(!do_mob(user, target, 10))
+			to_chat(user, "<span class='warning'>Our awakening of [target] has been interrupted!</span>")
+			hivehost.isintegrating = FALSE
+			return
+	hivehost.isintegrating = FALSE
+	var/datum/antagonist/hivevessel/V = new /datum/antagonist/hivevessel()
+	hivehost.avessels += target.mind
+	V.master = hivehost
+	V.hiveID = hivehost.hiveID
+	target.mind.add_antag_datum(V)
+	flash_color(user, flash_color="#800080", flash_time=10)
+	to_chat(user,"<span class='assimilator'>This vessel is now an extension of our will.</span>")
+	if(hivehost.unlocked_dominance)
+		target.add_overlay(hivehost.glow)
+
 /obj/effect/proc_holder/spell/self/hive_comms/cast(mob/living/user = usr)
 	var/message = stripped_input(user, "What do you want to say?", "Hive Communication")
+	var/datum/antagonist/hivemind/hivehost = user.mind.has_antag_datum(/datum/antagonist/hivemind)
+	if(!hivehost)
+		return
 	if(!message)
 		return
-	var/title = "One Mind"
-	var/span = "changeling"
-	if(user.mind && user.mind.has_antag_datum(/datum/antagonist/hivemind))
-		span = "assimilator"
-	var/my_message = "<span class='[span]'><b>[title] [findtextEx(user.name, user.real_name) ? user.name : "[user.real_name] (as [user.name])"]:</b> [message]</span>"
+	var/title = "Hive"
+	var/my_message = "<span class='changeling'><b>[title] [hivehost.hiveID]:</b> [message]</span>"
 	for(var/i in GLOB.player_list)
 		var/mob/M = i
-		if(is_hivehost(M) || is_hivemember(M))
+		if(is_hivehost(M))
 			to_chat(M, my_message)
 		else if(M in GLOB.dead_mob_list)
 			var/link = FOLLOW_LINK(M, user)
 			to_chat(M, "[link] [my_message]")
 
 	user.log_talk(message, LOG_SAY, tag="hive")
+
+
+/obj/effect/proc_holder/spell/target_hive/hive_shatter
+	name = "Crush Protections"
+	desc = "We destroy any Mindshield implants a vesssel might have, granting us further control over their mind."
+	action_icon_state = "shatter"
+
+	charge_max = 1800
+
+/obj/effect/proc_holder/spell/target_hive/hive_shatter/cast(list/targets, mob/living/user = usr)
+	var/mob/living/carbon/human/target = targets[1]
+	var/datum/antagonist/hivemind/hive = user.mind.has_antag_datum(/datum/antagonist/hivemind)
+	var/success = FALSE
+	if(!hive)
+		to_chat(user, "<span class='notice'>This is a bug. Error:HIVE1</span>")
+		return
+	if(!hive.hivemembers)
+		return
+
+	if(HAS_TRAIT(target, TRAIT_MINDSHIELD))
+		if(!do_after(user,200,0))
+			for(var/obj/item/implant/mindshield/M in target.implants)
+				to_chat(user, "<span class='notice'>We shatter their mental protections!</span>")
+				to_chat(target, "<span class='assimilator'>We feel a pang of pain course through our head!</span>")
```

Comment:
Should this be "we" and assimilator span? They're not an awakened vessel as the target, right?